scripts/metadata_comparison/test/lib/test_operation_ids.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+import unittest
+import json
+from metadata_comparison.lib.operation_ids import *
+from metadata_comparison.extractor import find_operation_ids_in_metadata
+from test.lib.helper_functions import read_resource
```

Comment:
I have checked out this read_resource statement:

```
    def read_resource(self, filename):
        path = f'test/resources/{filename}'
        with open(path, 'r') as file:
            data = file.read()
        return data
```

A few comments.
* `test/resources/{filename}` is dependent from where you run the tests. There is a `__file__` variable that in python means this exact file. So you can import relative to that. 
* pathlib is your friend. There is a library called `pathlib` from which you can import `Path`. In this case `path = Path(__file__).parent / Path("resources", filename)` creates a path that is always correct no matter where you run the test.
* pathlib is your friend again. `path` is now a `Path` object, which has a `read_text()` method, which does exactly what you do here. So `return path.read_text()` and it is done.
* Frankly this makes the whole `read_resource` function redundant. As you could also use a `RESOURCES` constant that is a pathlib.Path object. You can import this in the rest of the test code and do stuff like: `Path(RESOURCES, 'forkjoin_metadata_papi_v1alpha2.json').read_text()`. 

I always use pathlib in tests because of this. Also, it is OS agnostic as an added bonus.

Reply:
Much nicer!